Bug: worktree creation failure incorrectly increments builder systematic failure counter

## Problem

When the shepherd fails to create a worktree because the branch is already checked out in another worktree (an infrastructure error), it increments the \`builder_unknown_failure\` systematic failure counter as if the builder itself had failed. After 3 such increments the issue is moved to \`loom:blocked\`.

## Observed Behavior

\`\`\`
[ERROR] failed to create worktree: Preparing worktree (checking out 'feature/issue-2853')
fatal: 'feature/issue-2853' is already used by worktree at '/Users/rwalters/GitHub/loom'
ERROR: Failed to create worktree
[INFO] Label cleanup: reverted issue #2853 from loom:building to loom:issue
Systematic failure detected: pattern=builder_unknown_failure, threshold=3
[WARN] Systematic failure escalation: issue #2853 moved to loom:blocked (pattern=builder_unknown_failure, count=3)
\`\`\`

All 3 systematic failures for issue #2853 were worktree conflicts — the builder was never even launched.

## Root Cause

The shepherd's worktree creation failure falls through to the generic \`builder_unknown_failure\` path rather than being classified as an infrastructure error and excluded from the systematic failure counter.

## Expected Behavior

Infrastructure failures (worktree conflict, git errors before the builder starts) should:
1. Not increment the systematic failure counter
2. Emit a distinct error class (e.g. \`worktree_conflict\`) for diagnostics
3. Ideally provide actionable recovery guidance (e.g. which worktree holds the branch)

## References

- Observed during shepherd run for #2853 (2026-02-19)
- Related: #2916 (checkpoint commits on main caused the worktree conflict)

--- a/loom-tools/src/loom_tools/common/issue_failures.py
+++ b/loom-tools/src/loom_tools/common/issue_failures.py
@@ -48,13 +48,14 @@ ERROR_CLASS_BLOCK_THRESHOLDS: dict[str, int] = {
 }
 
 # Infrastructure error classes that should NOT count toward auto-blocking.
-# These represent environment/platform failures (MCP server down, auth timeout),
-# not problems with the issue itself.  Blocking an issue for infrastructure
-# failures is counterproductive — the issue would succeed once infrastructure
-# recovers.  See issue #2772.
+# These represent environment/platform failures (MCP server down, auth timeout,
+# worktree branch conflicts), not problems with the issue itself.  Blocking an
+# issue for infrastructure failures is counterproductive — the issue would
+# succeed once infrastructure recovers.  See issue #2772, #2918.
 INFRASTRUCTURE_ERROR_CLASSES: frozenset[str] = frozenset({
     "auth_infrastructure_failure",
     "mcp_infrastructure_failure",
+    "worktree_conflict",
 })
 
 # Backoff schedule: attempt -> iterations to skip

--- a/loom-tools/src/loom_tools/shepherd/cli.py
+++ b/loom-tools/src/loom_tools/shepherd/cli.py
@@ -1006,6 +1006,13 @@ def orchestrate(ctx: ShepherdContext) -> int:
                         exit_code = ShepherdExitCode.SYSTEMIC_FAILURE
                     elif result.data.get("rate_limit_abort"):
                         exit_code = ShepherdExitCode.RATE_LIMIT_ABORT
+                    elif result.data.get("worktree_conflict"):
+                        # Branch already checked out in another worktree —
+                        # infrastructure failure, not an issue with the code.
+                        # Use SYSTEMIC_FAILURE so _record_fallback_failure maps
+                        # it to an infrastructure error class excluded from the
+                        # systematic failure counter.  See issue #2918.
+                        exit_code = ShepherdExitCode.SYSTEMIC_FAILURE
                     else:
                         exit_code = ShepherdExitCode.BUILDER_FAILED
                     # Record abandonment details so _post_fallback_failure_comment
@@ -2231,6 +2238,21 @@ def _post_fallback_failure_comment(ctx: ShepherdContext, exit_code: int) -> None
                 "The issue has been returned to `loom:issue`. "
                 "Retry after the rate limit resets (usually a few minutes)."
             )
+        elif failure_data.get("worktree_conflict"):
+            error_detail = failure_data.get("error_detail", "")
+            failure_mode = (
+                "worktree branch conflict — the feature branch is already "
+                "checked out in another worktree (infrastructure failure, "
+                "not a defect in the issue)"
+            )
+            safe_to_retry = True
+            detail_line = f"\n\nGit error: `{error_detail}`" if error_detail else ""
+            advice = (
+                "The issue has been returned to `loom:issue` and is safe to retry. "
+                "To resolve: run `loom-clean --force` or `git worktree prune` "
+                "to remove the stale worktree, then retry the issue."
+                f"{detail_line}"
+            )
         elif failure_data.get("auth_failure"):
             failure_mode = (
                 "auth pre-flight failure — authentication check timed out "
@@ -2340,8 +2362,25 @@ def _record_fallback_failure(ctx: ShepherdContext, exit_code: int) -> None:
 
     # Classify the failure based on exit code
     if exit_code == ShepherdExitCode.SYSTEMIC_FAILURE:
-        error_class = "auth_infrastructure_failure"
-        details = "Builder failed due to auth/API infrastructure issue (fallback cleanup)"
+        # Check if this SYSTEMIC_FAILURE was actually a worktree branch conflict
+        # (branch already checked out in another worktree).  abandonment_info is
+        # set by the orchestrator before returning the exit code.  See #2918.
+        # Explicitly check isinstance(dict) to avoid false-positive from MagicMock
+        # in tests where abandonment_info is not set.
+        _ainfo = ctx.abandonment_info
+        _abandonment_data = _ainfo.get("failure_data", {}) if isinstance(_ainfo, dict) else {}
+        if _abandonment_data.get("worktree_conflict"):
+            error_class = "worktree_conflict"
+            error_detail = _abandonment_data.get("error_detail", "")
+            details = (
+                f"Worktree creation failed: branch already checked out in another "
+                f"worktree (infrastructure failure, not an issue defect)"
+            )
+            if error_detail:
+                details = f"{details} — {error_detail}"
+        else:
+            error_class = "auth_infrastructure_failure"
+            details = "Builder failed due to auth/API infrastructure issue (fallback cleanup)"
     elif exit_code == ShepherdExitCode.WORKTREE_ESCAPE:
         error_class = "builder_worktree_escape"
         details = "Builder escaped worktree and modified main instead (fallback cleanup)"

--- a/loom-tools/src/loom_tools/shepherd/phases/builder.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/builder.py
@@ -435,11 +435,20 @@ class BuilderPhase:
                 msg = "failed to create worktree"
                 if detail:
                     msg = f"{msg}: {detail}"
+                # Detect worktree branch conflicts: when the branch is already
+                # checked out in another worktree git raises a "already used by
+                # worktree" fatal error.  This is an infrastructure failure
+                # (git state, not issue quality) and must NOT count toward the
+                # systematic failure counter.  See issue #2918.
+                is_conflict = "already used by worktree" in detail
                 return PhaseResult(
                     status=PhaseStatus.FAILED,
                     message=msg,
                     phase_name="builder",
-                    data={"error_detail": detail},
+                    data={
+                        "error_detail": detail,
+                        "worktree_conflict": is_conflict,
+                    },
                 )
 
         # Create marker to prevent premature cleanup

--- a/loom-tools/tests/shepherd/test_cli.py
+++ b/loom-tools/tests/shepherd/test_cli.py
@@ -4685,6 +4685,106 @@ class TestRecordFallbackFailure:
         mock_record.assert_not_called()
         mock_detect.assert_not_called()
 
+    @patch("loom_tools.shepherd.cli.get_pr_for_issue", return_value=None)
+    @patch("loom_tools.common.systematic_failure.detect_systematic_failure", return_value=None)
+    @patch("loom_tools.common.systematic_failure.record_blocked_reason")
+    def test_records_worktree_conflict_class(
+        self,
+        mock_record: MagicMock,
+        mock_detect: MagicMock,
+        mock_get_pr: MagicMock,
+    ) -> None:
+        """Should record worktree_conflict (not auth_infrastructure_failure) when
+        abandonment_info identifies the failure as a branch-in-use conflict.
+        See issue #2918."""
+        ctx = MagicMock()
+        ctx.config.issue = 42
+        ctx.repo_root = Path("/fake/repo")
+        ctx.abandonment_info = {
+            "phase": "builder",
+            "exit_code": 9,
+            "failure_data": {
+                "worktree_conflict": True,
+                "error_detail": (
+                    "fatal: 'feature/issue-42' is already used by worktree at "
+                    "'/Users/user/GitHub/loom'"
+                ),
+            },
+            "message": "failed to create worktree",
+            "task_id": "abc123",
+        }
+
+        _record_fallback_failure(ctx, ShepherdExitCode.SYSTEMIC_FAILURE)
+
+        mock_record.assert_called_once()
+        call_args = mock_record.call_args
+        assert call_args.args[0] == Path("/fake/repo")
+        assert call_args.args[1] == 42
+        assert call_args.kwargs["error_class"] == "worktree_conflict"
+        assert call_args.kwargs["phase"] == "builder"
+        assert "already checked out in another worktree" in call_args.kwargs["details"]
+        # Error detail should be included in the details
+        assert "feature/issue-42" in call_args.kwargs["details"]
+        mock_detect.assert_called_once_with(Path("/fake/repo"))
+
+    @patch("loom_tools.shepherd.cli.get_pr_for_issue", return_value=None)
+    @patch("loom_tools.common.systematic_failure.detect_systematic_failure", return_value=None)
+    @patch("loom_tools.common.systematic_failure.record_blocked_reason")
+    def test_worktree_conflict_without_error_detail(
+        self,
+        mock_record: MagicMock,
+        mock_detect: MagicMock,
+        mock_get_pr: MagicMock,
+    ) -> None:
+        """worktree_conflict error class is recorded even without error_detail."""
+        ctx = MagicMock()
+        ctx.config.issue = 55
+        ctx.repo_root = Path("/fake/repo")
+        ctx.abandonment_info = {
+            "phase": "builder",
+            "exit_code": 9,
+            "failure_data": {
+                "worktree_conflict": True,
+                # No error_detail key
+            },
+            "message": "failed to create worktree",
+            "task_id": "xyz789",
+        }
+
+        _record_fallback_failure(ctx, ShepherdExitCode.SYSTEMIC_FAILURE)
+
+        call_args = mock_record.call_args
+        assert call_args.kwargs["error_class"] == "worktree_conflict"
+        # Details should not contain the error_detail suffix
+        assert "—" not in call_args.kwargs["details"]
+
+    @patch("loom_tools.shepherd.cli.get_pr_for_issue", return_value=None)
+    @patch("loom_tools.common.systematic_failure.detect_systematic_failure", return_value=None)
+    @patch("loom_tools.common.systematic_failure.record_blocked_reason")
+    def test_systemic_failure_without_worktree_conflict_stays_auth(
+        self,
+        mock_record: MagicMock,
+        mock_detect: MagicMock,
+        mock_get_pr: MagicMock,
+    ) -> None:
+        """SYSTEMIC_FAILURE without worktree_conflict flag stays auth_infrastructure_failure."""
+        ctx = MagicMock()
+        ctx.config.issue = 42
+        ctx.repo_root = Path("/fake/repo")
+        # Set abandonment_info without worktree_conflict
+        ctx.abandonment_info = {
+            "phase": "builder",
+            "exit_code": 9,
+            "failure_data": {"auth_failure": True},
+            "message": "auth timed out",
+            "task_id": "abc123",
+        }
+
+        _record_fallback_failure(ctx, ShepherdExitCode.SYSTEMIC_FAILURE)
+
+        call_args = mock_record.call_args
+        assert call_args.kwargs["error_class"] == "auth_infrastructure_failure"
+
 
 class TestCleanupFallbackIntegration:
     """Test that _cleanup_labels_on_failure calls the new fallback helpers (issue #2525)."""

--- a/loom-tools/tests/test_systematic_failure.py
+++ b/loom-tools/tests/test_systematic_failure.py
@@ -261,6 +261,41 @@ class TestDetectSystematicFailure:
         result = detect_systematic_failure(repo)
         assert result is None
 
+    def test_worktree_conflict_not_counted(self, repo: pathlib.Path) -> None:
+        """Consecutive worktree_conflict failures should NOT trigger systematic failure
+        detection, because they are infrastructure failures (git state), not issue
+        defects.  See issue #2918."""
+        _write_state(
+            repo,
+            {
+                "recent_failures": [
+                    {"error_class": "worktree_conflict"},
+                    {"error_class": "worktree_conflict"},
+                    {"error_class": "worktree_conflict"},
+                ]
+            },
+        )
+        result = detect_systematic_failure(repo)
+        assert result is None
+
+    def test_worktree_conflict_mixed_with_other_failures(self, repo: pathlib.Path) -> None:
+        """worktree_conflict failures are filtered out and don't inflate the count
+        for non-infrastructure failures.  See issue #2918."""
+        _write_state(
+            repo,
+            {
+                "recent_failures": [
+                    {"error_class": "builder_stuck"},
+                    {"error_class": "worktree_conflict"},
+                    {"error_class": "builder_stuck"},
+                    {"error_class": "worktree_conflict"},
+                    # Only 2 non-infra failures — below threshold of 3
+                ]
+            },
+        )
+        result = detect_systematic_failure(repo)
+        assert result is None
+
     def test_infra_mixed_with_issue_failures(self, repo: pathlib.Path) -> None:
         """Infra failures interspersed with issue failures don't inflate the count."""
         _write_state(

